src/core/CompUnit/Repository/Installation.pm

```diff
@@ -372,83 +372,68 @@ sub MAIN(:$name is copy, :$auth, :$ver, *@, *%) {
     }
 
     method script($file, :$name!, :$auth, :$ver) {
-        my $prefix = self.prefix;
-        my $lookup = $prefix.add('short').add(nqp::sha1($file));
-        return unless $lookup.e;
-
-        # Scripts using this interface could only have been installed long after the introduction of
-        # repo version 1, so we don't have to care about very old repos in this method.
-        my @dists = $lookup.dir.map({
-                my ($ver, $auth, $api, $resource-id) = $_.slurp.split("\n");
-                $resource-id ||= self!read-dist($_.basename)<files>{$file};
-                (id => $_.basename, ver => Version.new( $ver || 0 ), :$auth, :$api, :$resource-id).hash
-            }).grep({
-                $_.<auth> ~~ $auth
-                and $_.<ver> ~~ $ver
-            });
-        for @dists.sort(*.<ver>).reverse {
-            return self!resources-dir.add($_<resource-id>);
-        }
+        my $matching-scripts := self.files($file, :$name, :$auth, :$ver);
```

Comment:
I knew I was overlooking something obvious when my implementation made it appear redundant. Although after reexamining the method maybe we should get rid of `:$name`, which is unused.